### ISSUE TITLE
[Feature Request] @apollo/gateway - Expose the current request query path to the "willSendRequest" hook

### ISSUE THREAD

**Issue by talazenkot** [apollographql/federation#2353]
As far as I've researched, the path of the field currently resolved by the federation gateway is not exposed to any of the arguments passed to `willSendRequest` and `didReceiveResponse` hooks.
I've faced a scenario where I need to apply certain HTTP headers to the requests only sent under a specific node. 

![Screenshot 2023-01-25 at 18 23 41](https://user-images.githubusercontent.com/54143565/214618634-d26b0652-1341-4a6c-84a4-9eb97b5e9b93.png)
Given the image above, I would like to:

1. Resolve `node2` in some external apollo server
2. Respond with some header to the gateway
3. Make sure that the gateway passes this header to all children nodes (3, 4, 5, and their children) without its leaking into the sibling nodes (6 and its children).

I've managed to do that, only by caching this header for a certain path (the path from which it was received, in `didReceiveResponse`), and then in each request (`willSendRequest`), check if the current path is a children path of one of the cached paths, and if so, apply some logic, see the following code:
![Screenshot 2023-01-25 at 18 38 21](https://user-images.githubusercontent.com/54143565/214622564-5e5f5118-29f4-4f0e-ba6f-47d0e3fa77ea.png)

Bottom line, unless there's some more robust solution, I was wondering if it will be possible to expose the path of the current field.

**Comment by korinne** [comment]
Thanks for raising this request! Our team has added this to our backlog, and we will keep this issue updated once the request has been prioritized. We are also always open to community contributions.

**Comment by talazenkot** [comment]
@korinne Great! I will be glad to contribute and make this happen. Is there any other guidelines I need to follow except for the ones mentioned in https://github.com/apollographql/federation/blob/main/CONTRIBUTING.md#reporting-bugs ? Is there some sort of sync required?
